lib/src/utils/resolveRichTextToNodes.spec.ts

```diff
@@ -113,6 +113,83 @@ describe("resolveNode", () => {
     });
   });
 
+  it("text with textResolver", () => {
+    const node: SchemaNode = {
+      text: "Hello {name}",
+      type: "text",
+    };
+
+    const textResolver = (text: string) => {
+      return {
+        content: text.replace("{name}", "World"),
+      };
+    };
+
+    // default
+    expect(
+      resolveNode(node, {
+        textResolver,
+      })
+    ).toStrictEqual({
+      content: "Hello World",
+    });
+
+    // with marks
+    expect(
+      resolveNode(
+        {
+          ...node,
+          marks: [{ type: "bold" }],
+        },
+        {
+          textResolver,
+        }
+      )
+    ).toStrictEqual({
+      content: [
+        {
+          component: "b",
+          content: [
+            {
+              content: "Hello World",
+            },
+          ],
+        },
+      ],
+    });
+
+    // with schema override
+    expect(
+      resolveNode(node, {
+        schema: {
+          nodes: {
+            text: () => ({
+              component: "p",
+              props: { class: "class-1" },
+            }),
+          },
+        },
+        textResolver: (text) => ({
+          content: text.replace("{name}", "World"),
+          component: "span",
+          props: { class: "class-2" },
+        }),
```

Comment:
Hm, this does look weird honestly, and I am sure it clashes with the existing API. The problem is that conceptually the `textResolver` should be meant for text resolving only, and should not return `ComponentNode` at all. This function is not about the node logic at all, but rather about manipulating the `content` property in whatever the node is.

to get same result you have described with solution we should better have the following API:
```
     schema: {
        nodes: {
          text: () => ({
            component: "p",
            props: { class: "class-1" },
          }),
        },
      },
      textResolver: (text) => text.replace("{name}", "World")
```
(the textResolver extends the capability of the node, but does not override it)

Alternatively, we can expect the shape of `textResolver?: (str: string) => Pick<ComponentNode, 'content'>;`
and the usage:
```
 textResolver: (text) => ({
    content: text.replace("{name}", "World"),
 }),
```
but again, this may misslead users that it is not the whole component node but rather just a content, so might be bad alternative.

What do you think?

Reply:
it should return `ComponentNode` for cases when you example you want to support ICU syntax. So in this case all the texts need to be rendered inside some `.astro` component

I can suggest to support both outputs:
`textResolver?: (str: string) => ComponentNode | string;`